src/main/kotlin/org/rust/lang/core/psi/RsFile.kt

```diff
@@ -77,6 +77,12 @@ class RsFile(
             return cargoWorkspace?.isCrateRoot(file) ?: false
         }
 
+    override val isPublic: Boolean get() {
+        if (isCrateRoot) return true
+        val parentMod = `super` ?: return false
+        return parentMod.itemsAndMacros.any { it is RsModDeclItem && it.name == modName && it.isPublic }
```

Comment:
This does the same work twice, because `super` also has to find a `ModDecl`. Let's add something like `RsFile.declaration: RsModDecl?` and use it to implement both `super` and `isPublic`?

`org.rust.ide.navigation.goto.RsGotoSuperHandler` could also use this functionality.